src/test/java/teammates/test/cases/webapi/GetHasResponsesActionTest.java

```diff
@@ -0,0 +1,197 @@
+package teammates.test.cases.webapi;
+
+import java.util.List;
+
+import org.apache.http.HttpStatus;
+import org.testng.annotations.Test;
+
+import teammates.common.datatransfer.attributes.FeedbackQuestionAttributes;
+import teammates.common.datatransfer.attributes.FeedbackSessionAttributes;
+import teammates.common.datatransfer.attributes.InstructorAttributes;
+import teammates.common.exception.EntityDoesNotExistException;
+import teammates.common.util.Const;
+import teammates.ui.webapi.action.GetHasResponsesAction;
+import teammates.ui.webapi.action.JsonResult;
+import teammates.ui.webapi.output.HasResponsesData;
+import teammates.ui.webapi.output.MessageOutput;
+
+/**
+ * SUT: {@link GetHasResponsesAction}.
+ */
+public class GetHasResponsesActionTest extends BaseActionTest<GetHasResponsesAction> {
+
+    @Override
+    protected String getActionUri() {
+        return Const.ResourceURIs.HASRESPONSES;
+    }
+
+    @Override
+    protected String getRequestMethod() {
+        return GET;
+    }
+
+    @Test
+    @Override
+    protected void testExecute() throws Exception {
```

Comment:
Under test `execute`, you have tested the cases where a logged in instructor retrieves course data. What about the cases where a student tries to retrieve the course response details? (since any logged in user can access)